docs/preview/02-Guidance/Service-to-service Correlation/use-service-to-service-correlation-in-web-api.md

```diff
@@ -0,0 +1,310 @@
+* **Operation ID**: this ID describes a single operation within the transaction. This ID is used within a service to link all telemetry of that service together.
+* **Operation Parent ID**: this ID is create the parent/child link across services. When service A calls service B, then service A is the so called 'parent' of service B.
+
+The following list shows each step in the diagram:
+1. The initial call in this example doesn't contain any correlation headers. This can be seen as a first interaction call to a service. 
+2. Upon receiving at service A, the application will generate new correlation information. This correlation info will be used when telemetry is tracked on the service.
+3. When a call is made to service B, the **transaction ID** is sent but also the **operation parent ID** in the form of a hierarchical structure.
+4. The `jkl` part of this ID, describes the new parent ID for service B (when service B calls service C, then it will use `jkl` as parent ID)
+5. Service B responds to service A with the same information as the call to service B.
+6. The user receives both the **transaction ID** and **operation parent ID** in their final response.
+
+> Additional configuration is available to tweak this functionality, see the [dedicated Arcus Web API feature documentation](https://webapi.arcus-azure.net/features/correlation) for more in-depth information on HTTP correlation.
+
+## API Demonstration
+In this user-guide, two fictive API applications will be used to represent Service A and Service B of the diagram. Both services will be adapted to use service-to-service correlation.
+
+* **Product API** (Service A): receives an request to order a product if enough items of that product are in stock.
+* **Stock API** (Service B): response with the available items in stock of a given product
+
+The user interacts with the **Product API** to order their product. Internally, the **Product API** will contact the **Stock API** to verify if there are enough items of that product to complete the order. 
+
+> ⚠ Take into account that this sample should only be used for demonstration purposes and does not reflect a fully production-ready implementation. We have chosen to only provide the bare bones of the application so the changes are clear.
+
+### Product API: startup code
+First, lets look the initial code for the **Product API**. The startup code has the hosting and routing functionality to use API controllers. The configuration value `STOCK_API_URL` already reveals that this API will contact another API. This is also the reason why we added `.AddHttpClient()` to the startup code so we can create `HttpClient` instances. ([More information on creating HTTP clients in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-6.0)). 
```

Comment:
`.AddHttpClient` merely allows you to use HttpClient instances for dependency injection.  It allows you to inject HttpClient instances.

Reply:
Not sure if I understand this comment. Something you want to change here?

Reply:
I meant that you can always create HttpClient instances (via `new HttpClient()`).  Using the `AddHttpClient` method allows you to inject HttpClient instances into your controllers (or HttpClientFactory instances I guess).
So, I just thought the wording in that one sentence where you say that it allows you to 'create' instances is not entirely correct :)  You could change it to 'so we can inject HttpClient instances' . But it is no big deal.

/irrelvant for the discussion:
I mostly use the AddHttpClient different as well; in most cases I create a class which abstracts the calls to a specific API and that class gets injected a HttpClient:
```csharp
services.AddHttpClient<ProductApiProxy>();

...

public class ProductApiProxy
{
    ProductApiProxy( HttpClient http ) {}

   public Task<Product> GetProductAsync( int productId ) {}
}
```

Reply:
Ah in the examples, we use the `IHttpClientFactory` which then created the client (for named clients). So, was probably the reason to use 'create' here as the controller, in this case, 'creates' the client via the factory.

Abstracting the client with a specific implementation, yeah, I think that still works with our HTTP client correlation tracking as it enhances the client injected in your product-specific client.

Reply:
I'll change this from 'create' to 'inject'.

Reply:
Done in https://github.com/arcus-azure/arcus.observability/pull/446/commits/b2ddd5e1db06931183e67f28481f44033fa15dd3